mock/middle_ware.go

```diff
@@ -0,0 +1,27 @@
+package mock
```

Comment:
filename should be one word `middleware.go`

Reply:
Actually, what do you think about making this a subpackage of `http`? So this file would be in `http/mock/middleware.go`?